### DISCUSSION TITLE
Provide access to reflection probe data in shader language

### DISCUSSION THREAD

**Discussion by bertodelrio256** [godotengine/godot-proposals#9833] [3D]
I understand in godot 4 that lights, decals, and reflection probes are all clustered. In the godot shader language, you can override the light function by using void light() {}. for custom shading the default behavior of probes has been insuffient for customization. for example, say i dont want color from the probe on metals, or i want a specific fresnel curve or i want an intensity brighter then 1.0. it would be immensly useful to have something like:

```glsl

void reflections()
{
     ALBEDO *= PROBECOLOR.rgb * vec3(2.0) * skycolor.rgb;
}

```

**Comment by Calinou** [comment]
> or i want an intensity brighter then 1.0. 

See https://github.com/godotengine/godot-proposals/issues/7654, which has a pull request implementing it. You can also work around this by setting the ReflectionProbe's `intensity` property in a script, which bypasses the inspector hint's limits.
